### PR TITLE
feat(forms): add `daffFormsPasswordValidator`

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by griest024** [comment]
@damienwebdev I'm getting some strange type errors, we should talk about it tomorrow

File: libs/forms/src/validators/password.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  const minLength = (<DaffFormsPasswordValidatorLengthConfig>config).minLength;
+  const maxLength = (<DaffFormsPasswordValidatorLengthConfig>config).maxLength;
+
+  if (minLength) {
+    validators.push(Validators.minLength(minLength));
+  }
+  if (maxLength) {
+    validators.push(Validators.maxLength(maxLength));
+  }
+  if (classPattern.length > MATCH_EVERYTHING_REGEX.length) {
```
Comment (damienwebdev) [inline libs/forms/src/validators/password.ts]: This should likely be a strict equality, otherwise lgtm.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: libs/forms/src/validators/password.ts (hunk 1 above)
Comment (griest024) [reply libs/forms/src/validators/password.ts]: fixed

**Review by damienwebdev — APPROVED** [review]
Lgtm

**Pull request merged**